### ISSUE TITLE
[Release] v0.13.0 release schedule

### ISSUE THREAD

**Issue by ysh329** [apache/tvm#15134]
> Although [last released v0.12.0](https://github.com/apache/tvm/issues/14505) is on 17 May, the date of v0.12.0 branch cutting is 9 Apr. In other words, the v0.12.0 release is based on main branch of 9 Apr. It has been almost **three months** since April 9th.
> According to our [RFC-0067 about quarter-based releases](https://github.com/apache/tvm-rfcs/blob/main/rfcs/0067-quarterly-releases.md), I **hope to release v0.13.0 version in mid July**. It's meaningful especially for **AI infra companies which needs regularly merge with release version.**
> **In order to continue with our agreed release cycle as per discussion in [RFC#67](https://github.com/apache/tvm-rfcs/blob/main/rfcs/0067-quarterly-releases.md) , I'd like to propose a schedule for our next TVM release: `v0.13.0`.**

This schedule is tentative and may change as we progress through the process. In case dates change, this thread will be kept updated.

The proposed schedule is:

- [x] **1 July 2023** - a branch `v0.13.0` to be created from `main`
- [x] **1 July 2023** - a tag `v0.14.dev0` to be created, marking the beginning of the next development cycle
- [x] **2 July 2023** - modifying the version recorded to `v0.14.dev0` on `main` branch, https://github.com/apache/tvm/pull/15208 
- [x] **3 July 2023** - modifying the version recorded to `v0.13.0` on `v0.13.0` branch, ~https://github.com/apache/tvm/pull/15216~, https://github.com/apache/tvm/pull/15273
- [x] **7 July 2023** - cherry-picking soft deadline
  - during this time any reasonable change can be merged, comment on this issue with the commit you would like added to the release branch
- [x] **12 July 2023** - cherry-picking **hard deadline**
  - during this time only critical changes will be merged, comment on this issue with the commit you would like added to the release branch as well as the reasoning for adding it
- [x] **13 July 2023** - Draft release notes published, https://github.com/apache/tvm/issues/15295
- [x] **14 July 2023** - [prepare release candidate packaging v0.13.0.rc0 and uploading](https://tvm.apache.org/docs/contribute/release_process.html#upload-the-release-candidate)
  - Create tag v0.13.0.rc0 and Github release page: https://github.com/apache/tvm/releases/tag/v0.13.0.rc0
  - Upload to Apache repo: https://dist.apache.org/repos/dist/dev/tvm/tvm-v0.13.0-rc0/
- [x] **14 July 2023** - release voting opens, https://github.com/apache/tvm/issues/15313
- [x] **20 July 2023** - voting result, https://github.com/apache/tvm/issues/15376
- [x] **20 July 2023** - [upload the binaries to Apache mirrors and create a new release TAG](https://tvm.apache.org/docs/contribute/release_process.html#post-the-release) named v0.13.0 meanwhile remove the pre-release candidate TAG v0.13.0.rc0
- [x] **20 July 2023** - release date
- [x] **22 July 2023** - Update the TVM Website

**Call for release managers:** in case you want to be involved in upcoming releases, please manifest your interest in this thread and we'll try to organise. 😆 

See also:

- [Release steps](https://github.com/apache/tvm-rfcs/blob/main/rfcs/0067-quarterly-releases.md#reference-level-explanation)
- [TVM release process docs](https://tvm.apache.org/docs/contribute/release_process.html).

cc @apache/tvm-committers [@driazati](https://github.com/driazati) [@areusch](https://github.com/areusch) [@Mousius](https://github.com/Mousius) [@tqchen](https://github.com/tqchen) [@AndrewZhaoLuo](https://github.com/AndrewZhaoLuo)

**Comment by tqchen** [comment]
Happy to help with some of the branch cutting works, Thanks @ysh329 for volunteering

**Comment by Hzfengsy** [comment]
Thanks @ysh329. Happy to help with things that need permissions!

**Comment by spectrometerHBH** [comment]
I volunteer to help manage the release

**Comment by Johnson9009** [comment]
Can below two steps be merged into one? Because these steps have some relationship, i.e., the commit of "v0.14.dev0" must be the next commit of branch v0.13.0, in addition these two steps can be done quickly as long as the contributor has enough permission.

> 1 July 2023 - a branch v0.13.0 to be created from main
 3 July 2023 - a tag v0.14.dev0 to be created, marking the beginning of the next development cycle

In addition, there need add another step for modifying the version recorded in main branch code to "v0.14.dev0", because this step need to send a PR and need run all tests, so it maybe need 1~2 days to be merged.

**Comment by ysh329** [comment]
Please help create branch named v0.13.0 from main and v0.14.dev0 tag. @tqchen @Hzfengsy

**Comment by spectrometerHBH** [comment]
> 

The new branch and tag is now ready.

**Comment by ysh329** [comment]
Hi @spectrometerHBH, please make a PR on **v0.13.0 branch** like [this](https://github.com/apache/tvm/pull/14739/files), modify version to `0.13.0` **on v0.13.0 branch**.

**Comment by spectrometerHBH** [comment]
@ysh329 cc: https://github.com/apache/tvm/pull/15216

**Comment by ysh329** [comment]
Hi all, this PR https://github.com/apache/tvm/pull/15216 `about modifying the version recorded to v0.13.0 on v0.13.0 branch` is blocked by CI, but this won't block our `cherry-picking soft deadline` and `cherry-picking hard deadline`.

cc @driazati

**Comment by ysh329** [comment]
> Hi all, this PR #15216 `about modifying the version recorded to v0.13.0 on v0.13.0 branch` **is blocked by CI**, but this won't block our `cherry-picking soft deadline` and `cherry-picking hard deadline`.
> 
> cc @driazati

Can anybody help to see the CI issue? It seems these thing block the process of release schedule several times.

**Comment by ysh329** [comment]
> > Hi all, this PR #15216 `about modifying the version recorded to v0.13.0 on v0.13.0 branch` **is blocked by CI**, but this won't block our `cherry-picking soft deadline` and `cherry-picking hard deadline`.
> > cc @driazati
> 
> Can anybody help to see the CI issue? It seems these thing block the process of release schedule several times.

Hi all, this Jenkins CI issue is about our Jenkins config for those [non-main branchs will merge with latest main when CI running](https://github.com/apache/tvm/blob/main/ci/jenkins/generated/arm_jenkinsfile.groovy#L148-L154), this has no relation with Github branch protection.

Besides, this CI issue exposes our release step about version modification having some latent problem.

**Comment by tqchen** [comment]
I think the issue is that the PR always assumes merge with main in that Jenkinsfile

**Comment by ysh329** [comment]
> > > Hi all, this PR #15216 `about modifying the version recorded to v0.13.0 on v0.13.0 branch` **is blocked by CI**, but this won't block our `cherry-picking soft deadline` and `cherry-picking hard deadline`.
> > > cc @driazati
> > 
> > 
> > Can anybody help to see the CI issue? It seems these thing block the process of release schedule several times.
> 
> Hi all, this Jenkins CI issue is about our Jenkins config for those [non-main branchs will merge with latest main when CI running](https://github.com/apache/tvm/blob/main/ci/jenkins/generated/arm_jenkinsfile.groovy#L148-L154), this has no relation with Github branch protection.
> 
> Besides, this CI issue exposes our release step about version modification having some latent problem. We should modify version to 0.13.0 on main branch first and then do v0.13.0 branch cutting from main.:
> 
> * First, that can avoid conflict between main and non-main branch with CI, especially for one file in two branch existing divergence;
> * Second, because downstream companies codes are always from release branch, PR about version number is introduced on main but not release branch. Downstream companies can avoid conflicts about version number when git merge this release branch codes based on last release branch codes.

I think branch protection is no problem. Because The git merge operation is actually a protection mechanism for our publishing branch. Fundamentally, ensure that the PR introduced by the release branch is in the form of Cherry Pick on the main branch. And in theory, as long as the PR of cherry pick is in order, there will be no conflicts on the publishing branch.

The behavior of version modification on release branch is not in the form of Cherry Pick, and after branch cutting, we made changes on the main branch, which caused divergence thus conflicts. **I think we can first modify the version number on the main branch next time, because two reason as above**

**Comment by ysh329** [comment]
Hi all, considering the originally scheduled voting day is a weekend, I advance voting day to this Friday (07/14/2023).

**Comment by ysh329** [comment]
Hi all, v0.13.0 Release Candidate Notes published in advance: https://github.com/apache/tvm/issues/15295

**Comment by ysh329** [comment]
Please help [prepare release candidate packaging and uploading](https://tvm.apache.org/docs/contribute/release_process.html#upload-the-release-candidate) for voting.

cc @Hzfengsy @junrushao @tqchen

**Comment by ysh329** [comment]
> Please help [prepare release candidate packaging and uploading](https://tvm.apache.org/docs/contribute/release_process.html#upload-the-release-candidate) for voting.
> 
> cc @Hzfengsy @junrushao @tqchen

Please help delete branch `v0.13.0-rc`, `rc-v0.13.0`, and prepare candidate packaging for branch `v0.13.0`.

**Comment by Hzfengsy** [comment]
@ysh329 packages are ready now.

**Comment by ysh329** [comment]
## Check sha512
```shell
# Check sha512
$ shasum -a 512 ./apache-tvm-src-v0.13.0.rc0.tar.gz
da0fa8ad9a858b297fc1dc883a37505f6277e57568e9770a734b99ad34379d4bde6f7fde0d93a7ebf2454dd2086d9c1567e3768cd55bc50ca9749762628b89c0  ./apache-tvm-src-v0.13.0.rc0.tar.gz
$ cat ./apache-tvm-src-v0.13.0.rc0.tar.gz.sha512
da0fa8ad9a858b297fc1dc883a37505f6277e57568e9770a734b99ad34379d4bde6f7fde0d93a7ebf2454dd2086d9c1567e3768cd55bc50ca9749762628b89c0  apache-tvm-src-v0.13.0.rc0.tar.gz
```

## Check asc
```shell
# Check asc
$ gpg --import ./KEYS
gpg: key 6754EA97C55E3DEB: "Tianqi Chen (CODE SIGNING KEY) <tqchen@apache.org>" not changed
gpg: key 8955D84E20C9AB0C: "Yizhi Liu <javelinjs@gmail.com>" not changed
gpg: key 3A709F6F2B86E9D6: "Yizhi Liu (Yizhi Liu at Apache) <liuyizhi@apache.org>" not changed
gpg: key CA751254E97B9FE4: "Yizhi Liu <liuyizhi@apache.org>" not changed
gpg: key ED03B26E4FC3509F: "Ziheng Jiang <ziheng@apache.org>" not changed
gpg: key 9FBECBD3D75EFD4B: "Zhi Chen <zhic@apache.org>" not changed
gpg: key 35ABC9676004ADAE: "Junru Shao <junrushao@apache.org>" not changed
gpg: key 2C75E5A496C80880: "Wuwei Lin <wuwei@apache.org>" not changed
gpg: key 07FA463F1C926F48: "David Riazati <driazati@apache.org>" not changed
gpg: key C9A56ABD5CCA3EB8: "Andrew Zhao Luo <andrewzhaoluo@apache.org>" not changed
gpg: key 28D4862222B8EC31: "Leandro Nunes <leandron@apache.org>" not changed
gpg: key 06D051CA84EF3749: public key "Siyuan Feng (CODE SIGNING KEY) <syfeng@apache.org>" imported
gpg: Total number processed: 12
gpg:               imported: 1
gpg:              unchanged: 11
$ gpg --verify ./apache-tvm-src-v0.13.0.rc0.tar.gz.asc ./apache-tvm-src-v0.13.0.rc0.tar.gz
gpg: Signature made Fri Jul 14 10:24:35 2023 CST
gpg:                using RSA key 664EF29634C05669C3DCF83106D051CA84EF3749
gpg: Good signature from "Siyuan Feng (CODE SIGNING KEY) <syfeng@apache.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 664E F296 34C0 5669 C3DC  F831 06D0 51CA 84EF 3749

$ gpg --edit-key 664EF29634C05669C3DCF83106D051CA84EF3749
gpg> trust
Your decision? 5
Do you really want to set this key to ultimate trust? (y/N) y

$ gpg --verify ./apache-tvm-src-v0.13.0.rc0.tar.gz.asc ./apache-tvm-src-v0.13.0.rc0.tar.gz
gpg: Signature made Fri Jul 14 10:24:35 2023 CST
gpg:                using RSA key 664EF29634C05669C3DCF83106D051CA84EF3749
gpg: checking the trustdb
gpg: marginals needed: 3  completes needed: 1  trust model: pgp
gpg: depth: 0  valid:   3  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 3u
gpg: next trustdb check due at 2025-04-22
gpg: Good signature from "Siyuan Feng (CODE SIGNING KEY) <syfeng@apache.org>" [ultimate]
```

## Check binary
```shell
# check binary
$ find ./apache-tvm-src-v0.13.0 -type f -exec file {} + | grep -w "ELF\|shared object"
# find nothing
```

## Compile

**Comment by ysh329** [comment]
Hi all, vote starts (https://github.com/apache/tvm/issues/15313). Everyone is welcomed to vote. Please vote by replying to this thread explicitly. Vote will close July. 20th at 23:59M GMT.

**Comment by ysh329** [comment]
Hi all, vote passes with https://github.com/apache/tvm/issues/14844, next we'll [upload the binaries to Apache mirrors and create a new release TAG](https://tvm.apache.org/docs/contribute/release_process.html#post-the-release) named v0.13.0 meanwhile remove the pre-release candidate TAG v0.13.0.rc0. This program needs help please.

Besides, we need update doc website too. According to [last release talk](https://github.com/apache/tvm/issues/14505#issuecomment-1546528974_), this step would need ASF perm. 

cc @Hzfengsy @junrushao @vinx13 @tqchen

**Comment by ysh329** [comment]
> Hi all, vote passes with #14844, next we'll [upload the binaries to Apache mirrors and create a new release TAG](https://tvm.apache.org/docs/contribute/release_process.html#post-the-release) named v0.13.0 meanwhile remove the pre-release candidate TAG v0.13.0.rc0. This program needs help please.
> 
> Besides, we need update doc website too. According to [last release talk](https://github.com/apache/tvm/issues/14505#issuecomment-1546528974_) as below, this step would need ASF perm.
> 
> > @ysh329 Thanks for spotting this. The download page is broken now, it only includes links for v0.8 release. This is because previous PRs to tvm-site wasn't send to main branch, so manual rebuild of the site will overwrite the contents.
> > The update of the site should follow steps: 1) send PR to tvm-site/main to update download links and run [scripts/task_deploy_asf_site.sh](https://github.com/apache/tvm-site/blob/main/scripts/task_deploy_asf_site.sh) 2) push docs to asf-site branch
> 
> _Originally posted by @vinx13 in [#14505 (comment)](https://github.com/apache/tvm/issues/14505#issuecomment-1546528974)_
> 
> cc @Hzfengsy @junrushao @vinx13 @tqchen

Hi all, let's persevere until the end. The sight of a great triumph is coming soon. cc @Hzfengsy @junrushao @vinx13 @tqchen

**Comment by ysh329** [comment]
> > Hi all, vote passes with #14844, next we'll [upload the binaries to Apache mirrors and create a new release TAG](https://tvm.apache.org/docs/contribute/release_process.html#post-the-release) named v0.13.0 meanwhile remove the pre-release candidate TAG v0.13.0.rc0. This program needs help please.
> > Besides, we need update doc website too. According to [last release talk](https://github.com/apache/tvm/issues/14505#issuecomment-1546528974_) as below, this step would need ASF perm.
> > > @ysh329 Thanks for spotting this. The download page is broken now, it only includes links for v0.8 release. This is because previous PRs to tvm-site wasn't send to main branch, so manual rebuild of the site will overwrite the contents.
> > > The update of the site should follow steps: 1) send PR to tvm-site/main to update download links and run [scripts/task_deploy_asf_site.sh](https://github.com/apache/tvm-site/blob/main/scripts/task_deploy_asf_site.sh) 2) push docs to asf-site branch
> > 
> > 
> > _Originally posted by @vinx13 in [#14505 (comment)](https://github.com/apache/tvm/issues/14505#issuecomment-1546528974)_
> > cc @Hzfengsy @junrushao @vinx13 @tqchen
> 
> Hi all, let's persevere until the end. The sight of a great triumph is coming soon. cc @Hzfengsy @junrushao @vinx13 @tqchen

Hi @vinx13 , can you help update docs to website?

**Comment by vinx13** [comment]
I've updated the website
